### PR TITLE
[Feature] Add higher order operations

### PR DESCRIPTION
**Pull request by dominikandreasseitz**
Moving several key, higher-order block operations from qadence to pyqtorch.
Namely `Add`, `Scale` and `Merge`

### CODE REVIEW

File: tests/test_circuit.py [hunk 1]
```diff
@@ -136,3 +136,27 @@ def test_adjoint_duplicate_params() -> None:
         exp_adjoint, tuple(values.values()), torch.ones_like(exp_adjoint)
     )[0]
     assert torch.allclose(grad_ad, grad_adjoint)
+
+
+def test_scale() -> None:
```
Comment (Roland-djee) [inline tests/test_circuit.py:141]: Maybe expand the tests to more cases than the hardcoded ones.

**Review by Roland-djee — APPROVED** [review]
Thanks @dominikandreasseitz. LGTM. Could you please expand a bit on the need in a issue and link to this PR ?

File: tests/test_circuit.py [hunk 2]
```diff
@@ -136,3 +138,39 @@ def test_adjoint_duplicate_params() -> None:
...
+    scaledwf_primitive = pyq.Scale(gate, "scale")(state, values)
+    scaledwf_composite = pyq.Scale(pyq.Sequence([gate]), "scale")(state, values)
+    assert torch.allclose(wf, scaledwf_primitive)
+    assert torch.allclose(wf, scaledwf_composite)
+
+
+def test_add() -> None:
+    x = pyq.X(0)
+    z = pyq.Z(1)
+    state = pyq.zero_state(2)
```
Comment (jpmoutinho) [inline tests/test_circuit.py:160]:
```suggestion
    state = pyq.random_state(2)
```

**Review by jpmoutinho — APPROVED** [review]

File: tests/test_circuit.py (hunk 1 above)
Comment (dominikandreasseitz) [reply tests/test_circuit.py]: did so

File: tests/test_circuit.py (hunk 2 above)
Comment (dominikandreasseitz) [reply tests/test_circuit.py:160]: added

**Pull request merged**